src/__screenshot_tests__/form-fields-screenshot-test.tsx

```diff
@@ -27,6 +27,27 @@ test.each(TESTABLE_DEVICES)('Default textfield appears properly on %s', async (d
     expect(image).toMatchImageSnapshot();
 });
 
+test('Textfield collapses its label when autofill fills out the form', async () => {
```

Comment:
Even though the fix is applied on the `TextFieldBase` I'd test just a component for this corner case. WDYT?